src/xdg-autostart-generator/xdg-autostart-service.c

```diff
@@ -0,0 +1,655 @@
+
+        first_arg = 1;
+        for (i = n = 0; exec_split[i]; i++) {
+                _cleanup_free_ char *orig = NULL;
+                _cleanup_free_ char *c = NULL;
+                _cleanup_free_ char *raw = NULL;
+                _cleanup_free_ char *p = NULL;
+                _cleanup_free_ char *escaped = NULL;
+                _cleanup_free_ char *quoted = NULL;
+
+                orig = TAKE_PTR(exec_split[i]);
+                r = cunescape(orig, 0, &c);
+                if (r < 0) {
+                        return log_debug_errno(r, "Failed to unescape '%s': %m", orig);
+                }
+
+                if (first_arg) {
+                        _cleanup_free_ char *executable = NULL;
+
+                        /* This is the executable, find it in $PATH */
+                        first_arg = 0;
+                        r = find_binary(c, &executable);
+                        if (r < 0) {
+                                return log_info_errno(r, "Exec binary '%s' does not exist: %m", c);
+                        }
+
+                        escaped = cescape(executable);
+                        if (!escaped)
+                                return log_oom();
+
+                        exec_split[n++] = TAKE_PTR(escaped);
+                        continue;
+                }
+
+                /*
+                 * Remove any standardised XDG fields; we assume they never appear as
+                 * part of another argument as that just does not make any sense as
+                 * they can be empty (GLib will e.g. turn "%f" into an empty argument).
+                 * Other implementations may handle this differently.
+                 */
+                if (STR_IN_SET(c,
+                               "%f", "%F",
+                               "%u", "%U",
+                               "%d", "%D",
+                               "%n", "%N",
+                               "%i", /* Location of icon, could be implemented. */
+                               "%c", /* Translated application name, could be implemented. */
+                               "%k", /* Location of desktop file, could be implemented. */
+                               "%v",
+                               "%m"
+                               ))
+                        continue;
+
+                /*
+                 * %% -> % and then % -> %% means that we correctly quote any %
+                 * and also quote any left over (and invalid) % specifier from
+                 * the desktop file.
+                 */
+                raw = strreplace(c, "%%", "%");
+                if (!raw)
+                        return log_oom();
+                p = strreplace(raw, "%", "%%");
+                if (!p)
+                        return log_oom();
+                escaped = cescape(p);
+                if (!escaped)
+                        return log_oom();
+
+                asprintf(&quoted, "\"%s\"", escaped);
+                if (!quoted)
```

Comment:
should probably better be `if (asprintf(…)  < 0` 

(the current code is ok too i guess)

Reply:
still unaddressed